dicetable.py

```diff
@@ -0,0 +1,327 @@
+
+
+class dicetable(object):
+    def __init__(self,dsize):
+        self.totaldice = 0
+        self.table = {0:1}
+        self.dsize = dsize
+        
+
+    def getTableVal(self,value):
+        return self.table.get(value,0)
+    def getTable(self):
+        return self.table
+    def getDsize(self):
+        return self.dsize
+    def getTotalDice(self):
+        '''returns how many dice were used to generate the values in your table'''
+        return self.totaldice
+    def max(self):
+        '''returns the highest die roll in table'''
+        return self.dsize*self.totaldice
+    def min(self):
+        '''returns the lowest die roll in table'''
+        return self.totaldice
+    
+    def addADie(self):
+        '''takes your dicetable object, and calculates all the new combinations
+        if you add a new die of the dsize of the object'''
+        #so for d3, this would take {0:1,1:2} and 
+        #update to {1:1,2:1,3:1}+{2:2,3:2,4:2}
+        self.totaldice+=1
+        newdic = {}
+        for el in self.table:
+            currentval = self.table[el]
+            for x in range(self.dsize):
+                
+                newdic[el+x+1] = (newdic.get(el+x+1,0)+currentval)
+        self.table= newdic
+     
+    def __str__(self):
+        return str(self.totaldice)+'D'+str(self.dsize)    
+    def printTable(self):
+        '''go on. give it a try.  you'll never guess what this function does.
+        it's a surprise'''
+        for el in range(self.min(), self.max()+1):
+            if el<10:
+                print ' '+str(el)+':'+str(self.table[el])
+            else:
+                print(str(el)+':'+str(self.table[el]))  
+                
+    def mean(self):
+        '''i mean, don't you just sometimes look at a table of values
+        and wonder what the mean is?'''
+        return (self.totaldice*(1+self.dsize)/2.0) 
+    def stddev(self):
+        '''wassamatter you! you don't know what standard deviation is?'''
+        try:
+            avg = self.mean()
+            sqs = 0
+            count = 0
+            for x in self.table.keys():
+                sqs += self.table[x]*((avg - x)**2)
+                count += self.table[x]
+            return round((sqs/count)**0.5,4)
```

Comment:
perhaps you should use decimal https://docs.python.org/2/library/decimal.html

Also, something we need to talk about it don't repeat yourself.